File: src/main/java/org/apache/fineract/client/services/RescheduleLoansApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import io.reactivex.Observable;
import io.reactivex.Completable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;


import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface RescheduleLoansApi {
  /**
   * 
   * 
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("rescheduleloans")
  Observable<String> createLoanRescheduleRequest(
    @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @param scheduleId  (required)
   * @param command  (optional)
   * @return Observable&lt;String&gt;
   */
  @GET("rescheduleloans/{scheduleId}")
  Observable<String> readLoanRescheduleRequest(
    @retrofit2.http.Path("scheduleId") Long scheduleId, @retrofit2.http.Query("command") String command
  );

  /**
   * 
   * 
   * @param command  (optional)
   * @return Observable&lt;String&gt;
   */
  @GET("rescheduleloans")
  Observable<String> retrieveAllRescheduleRequest(
    @retrofit2.http.Query("command") String command
  );

  /**
   * 
   * 
   * @return Observable&lt;String&gt;
   */
  @GET("rescheduleloans/template")
  Observable<String> retrieveTemplate9();
    

  /**
   * 
   * 
   * @param scheduleId  (required)
   * @param command  (optional)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("rescheduleloans/{scheduleId}")
  Observable<String> updateLoanRescheduleRequest(
    @retrofit2.http.Path("scheduleId") Long scheduleId, @retrofit2.http.Query("command") String command, @retrofit2.http.Body String body
  );

}

File: src/main/kotlin/org/openapitools/client/apis/RescheduleLoansApi.kt
package org.openapitools.client.apis

import org.openapitools.client.models.GetLoanRescheduleRequestResponse
import org.openapitools.client.models.GetRescheduleReasonsTemplateResponse
import org.openapitools.client.models.PostCreateRescheduleLoansRequest
import org.openapitools.client.models.PostCreateRescheduleLoansResponse
import org.openapitools.client.models.PostUpdateRescheduleLoansRequest
import org.openapitools.client.models.PostUpdateRescheduleLoansResponse
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface RescheduleLoansApi {
    /**
     * Create loan reschedule request
     * Create a loan reschedule request.
     * Responses:
     *  - 200: OK
     *
     * @param postCreateRescheduleLoansRequest
     * @return [PostCreateRescheduleLoansResponse]
     */
    @POST("v1/rescheduleloans")
    suspend fun createLoanRescheduleRequest(@Body postCreateRescheduleLoansRequest: PostCreateRescheduleLoansRequest): PostCreateRescheduleLoansResponse

    /**
     * Retrieve loan reschedule request by schedule id
     * Retrieve loan reschedule request by schedule id
     * Responses:
     *  - 200: OK
     *
     * @param scheduleId
     * @param command  (optional)
     * @return [GetLoanRescheduleRequestResponse]
     */
    @GET("v1/rescheduleloans/{scheduleId}")
    suspend fun readLoanRescheduleRequest(
        @Path("scheduleId") scheduleId: Long,
        @Query("command") command: String? = null
    ): GetLoanRescheduleRequestResponse

    /**
     * Retrieve all reschedule requests
     * Retrieve all reschedule requests.
     * Responses:
     *  - 200: OK
     *
     * @param command  (optional)
     * @param loanId  (optional)
     * @return [kotlin.collections.List<GetLoanRescheduleRequestResponse]
     */
    @GET("v1/rescheduleloans")
    suspend fun retrieveAllRescheduleRequest(
        @Query("command") command: String? = null,
        @Query("loanId") loanId: Long? = null
    ): List<GetLoanRescheduleRequestResponse>

    /**
     * Retrieve all reschedule loan reasons
     * Retrieve all reschedule loan reasons as a template
     * Responses:
     *  - 200: OK
     *
     * @return [GetRescheduleReasonsTemplateResponse]
     */
    @GET("v1/rescheduleloans/template")
    suspend fun retrieveTemplate10(): GetRescheduleReasonsTemplateResponse

    /**
     * Update loan reschedule request
     * Update a loan reschedule request by either approving/rejecting it.
     * Responses:
     *  - 200: OK
     *
     * @param scheduleId
     * @param postUpdateRescheduleLoansRequest
     * @param command  (optional)
     * @return [PostUpdateRescheduleLoansResponse]
     */
    @POST("v1/rescheduleloans/{scheduleId}")
    suspend fun updateLoanRescheduleRequest(
        @Path("scheduleId") scheduleId: Long,
        @Body postUpdateRescheduleLoansRequest: PostUpdateRescheduleLoansRequest,
        @Query("command") command: String? = null
    ): PostUpdateRescheduleLoansResponse

}
